seisgo/noise.py

```diff
@@ -384,55 +379,218 @@ def do_correlation(sfile,win_len,step,maxlag,cc_method='xcorr',acorr_only=False,
+                #fftdata=[]
+                #get source raw data
+                tt1=time.time()
+                source=assemble_raw(sfile,ds,sall_tags,sloc,issta,sinv1,v,correct_orientation=correct_orientation,pad_thre=pad_thre,do_rotation=do_rotation)
+                tt2=time.time()
+                ttt2+=tt2-tt1
+                #print('##################################################### Assemble source raw data takes {:.5f}'.format(tt2-tt1))
+                if source is None:
+                    print('CONTINUE! Something wrong when assembling raw data.')
+                    #sys.exit()
+                    continue
+                
+                data_cache[issta]=source
+            
+            # if rotation is True, loop through all receiver stations, do rotation and then do fft and cross-correlation
+            if do_rotation:
+                iend=len(sta_list)
+                istart=iiS
+                src=issta
+                # loop through all receiver stations
+                for iiR in range(istart,iend):
+                    rcv=sta_list[iiR]
+                    irsta=rcv
+                    print('receiver: {}'.format(irsta))
+                    
+                    if irsta in data_cache:
+                        receiver=data_cache[irsta]
+                    else:
+                        rall_tags = ds.waveforms[irsta].get_waveform_tags()
+                        rloc=loc[iiR][0]
+                        rinv1=loc[iiR][1]
+                        # get receiver raw data
+                        tt11=time.time()
+                        receiver=assemble_raw(sfile,ds,rall_tags,rloc,irsta,rinv1,v,correct_orientation=correct_orientation,pad_thre=pad_thre,do_rotation=do_rotation)
+                        tt12=time.time()
+                        ttt6+=tt12-tt11
+                        if receiver is None:
+                            print('CONTINUE! Something wrong when assembling raw data.')
+                            continue
+                        
+                        data_cache[irsta]=receiver
+                    
+                    if (acorr_only and src==rcv) or (xcorr_only and src != rcv) or (not acorr_only and not xcorr_only):
+                        #if v:print('receiver: %s' % (rcv))
+                        
+                        # perform rotation, the output should be a nested list which contains at most 9 lists, of which each contains two obspy.trace objects.
+                        tt5=time.time()
+                        tr_pairs=rotation(source,receiver,channel_pairs)
+                        tt6=time.time()
+                        ttt3+=tt6-tt5
+                        #print('##################################################### Channel rotation takes {:.5f}'.format(tt6-tt5))
+
+                        if tr_pairs is None:
+                            print('Station pair {0}-{1} can not be rotated, proceed to the next receiver'.format(src,rcv))
```

Comment:
more specific in error message. e.g., tr_pairs is empty after rotation.

Reply:
rotation() will only return None when either source or receiver has less than 2 horizontal channels. The corresponding error message will be printed if any of the cases is true.
The error message you mentioned is mainly for specifying which station pair has problem.